### PR TITLE
Configuring devise_token_auth

### PR DESCRIPTION
**Pull request by SebasWolox**
## Summary
Had to do rollback in two steps due destroying User table and then had to create new migration that includes new users columns that gem's needs.

Was tested with Postman at this endpoint: 'localhost:3000/users/confirmation/new' responding with status 406 due does not have controller and method response. 

## Trello Card
https://trello.com/c/9Mq1UYIN/24-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -65,4 +65,7 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 gem 'rubocop', '~> 0.59.2', require: false
 
 #devise
-gem 'devise'
\ No newline at end of file
+gem 'devise'
+
+gem 'devise_token_auth'
+gem 'omniauth', '~> 1.6', '>= 1.6.1'
```
Comment (mtejedorwolox) [inline Gemfile]: add newline

Comment (mtejedorwolox) [reply Gemfile]: why omniauth?

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  # Include default devise modules.
+  devise  :database_authenticatable, :registerable, :recoverable, :rememberable,
+          :trackable, :validatable, :confirmable, :omniauthable
```
Comment (mtejedorwolox) [inline app/models/user.rb]: why confirmable and omniauthable?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
   devise_for :users
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+   # token auth routes available at /api/v1/auth
+  namespace :api do
```
Comment (mtejedorwolox) [inline config/routes.rb]: use gem versionist method `api_version`

File: db/migrate/20181010151538_devise_token_auth_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
+class DeviseTokenAuthCreateUsers < ActiveRecord::Migration[5.2]
+  def change
+    
+    create_table(:users) do |t|
+      ## Required
+      t.string :provider, :null => false, :default => "email"
```
Comment (mtejedorwolox) [inline db/migrate/20181010151538_devise_token_auth_create_users.rb]: remove provider and uid

File: Gemfile (hunk 1 above)
Comment (SebasWolox) [reply Gemfile]: Cuando se inicializó devise_token_auth, lo pedía

File: db/migrate/20181010151538_devise_token_auth_create_users.rb (hunk 4 above)
Comment (SebasWolox) [reply db/migrate/20181010151538_devise_token_auth_create_users.rb]: toca tirar rollback? tengo la base de datos con lo ultimo que he estado haciendo

File: app/models/user.rb (hunk 2 above)
Comment (SebasWolox) [reply app/models/user.rb]: cuando se inicializó devise_token_auth, lo dejé porque en el card no decía que no.

File: db/migrate/20181010151538_devise_token_auth_create_users.rb (hunk 4 above)
Comment (SebasWolox) [reply db/migrate/20181010151538_devise_token_auth_create_users.rb]: screen shot 2018-10-12 at 11 00 35 am

Al quitar esas dos columnas, no puedo crear un User ya que me pide, `provider`

**Pull request merged**